youtube_dl/utils.py

```diff
@@ -2512,6 +2514,84 @@ def short2full(cls, code):
         return cls._country_map.get(code.upper())
 
 
+def _load_exe_resource(res_type, res_name):
+    kernel32 = ctypes.windll.kernel32
+
+    exe_handle = 0  # NULL: Current process
+
+    res_info = kernel32.FindResourceW(exe_handle, res_name, res_type)
+    res_handle = kernel32.LoadResource(exe_handle, res_info)
+    res_data = kernel32.LockResource(res_handle)
+    res_len = kernel32.SizeofResource(exe_handle, res_info)
+    res_arr = ctypes.cast(res_data, ctypes.POINTER(ctypes.c_char))[:res_len]
+    return res_arr
+
+
+class I18N(object):
+    def __init__(self):
+        self.default_lang = None
+        self.domain = 'youtube_dl'
+
+    def translate(self, s):
+        if self.default_lang is not None:
+            lang = self.default_lang
+        else:
+            lang, _ = locale.getdefaultlocale()
+
+        try:
+            t = gettext.translation(self.domain, find_file_in_root('share/locale/'), [lang])
```

Comment:
When running as `youtube-dl.exe` build with python 2 from path containing non-ASCII `find_file_in_root` ends up returning `None`. When `localedir=None` is passed to `gettext.translation` it picks up `_default_localedir` using `os.path.join` with mixture of byte strings and unicode strings that results in similar problem:
```
C:\temp\тест\youtube-dl>youtube-dl.exe -vs test:youtube
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'-vs', u'test:youtube']
[debug] Encodings: locale cp1251, fs mbcs, out cp866, pref cp1251
[debug] youtube-dl version 2015.12.18
[debug] Python version 2.7.10 - Windows-7-6.1.7601-SP1
[debug] exe versions: ffmpeg N-73189-g7728d23, ffprobe N-73189-g7728d23, rtmpdump 2.4
[debug] Proxy map: {}
Traceback (most recent call last):
  File "__main__.py", line 19, in <module>
  File "youtube_dl\__init__.pyo", line 414, in main
  File "youtube_dl\__init__.pyo", line 404, in _real_main
  File "youtube_dl\YoutubeDL.pyo", line 1676, in download
  File "youtube_dl\YoutubeDL.pyo", line 664, in extract_info
  File "youtube_dl\extractor\common.pyo", line 291, in extract
  File "youtube_dl\extractor\testurl.pyo", line 65, in _real_extract
  File "youtube_dl\utils.pyo", line 2556, in translate
  File "gettext.pyo", line 468, in translation
  File "gettext.pyo", line 451, in find
  File "ntpath.pyo", line 85, in join
UnicodeDecodeError: 'ascii' codec can't decode byte 0xf2 in position 6: ordinal not in range(128)
```
We can workaround be skipping it completely when we found no locale dir:
```python
        locale_dir = find_file_in_root('share/locale/')
        if locale_dir:
            try:
                ...
```
Or we can mimic what `gettext` do by appending extra path in `get_root_dirs`:
```python
ret.append(os.path.join(decodeFilename(sys.prefix), 'share', 'locale'))
```